extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java

```diff
@@ -77,10 +78,11 @@ public SystemPropertyBuildItem setNettyMachineId() {
         return new SystemPropertyBuildItem("io.netty.machineId", nettyMachineId);
     }
 
-    @BuildStep
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
```

Comment:
This is needed because of the addition of `BuildProducer<SystemPropertyBuildItem> systemProperties`, right?